permiso/src/main/java/com/greysonparrelli/permiso/PermisoDialogFragment.java

```diff
@@ -36,16 +41,34 @@
      * @return A new {@link PermisoDialogFragment}.
      */
     public static PermisoDialogFragment newInstance(@Nullable String title, @NonNull String message, @Nullable String buttonText) {
-        PermisoDialogFragment dialogFragment = new PermisoDialogFragment();
+        return newInstance(new Builder()
+                        .setTitle(title)
+                        .setMessage(message)
+                        .setButtonText(buttonText));
+    }
+
+    private static PermisoDialogFragment newInstance(Builder builder) {
+        PermisoDialogFragment frag = new PermisoDialogFragment();
+        frag.setArguments(getDialogArgs(builder));
+        return frag;
+    }
 
-        // Build arguments bundle
+    private static Bundle getDialogArgs(Builder builder) {
         Bundle args = new Bundle();
-        args.putString(KEY_TITLE, title);
-        args.putString(KEY_MESSAGE, message);
-        args.putString(KEY_BUTTON_TEXT, buttonText);
-        dialogFragment.setArguments(args);
+        args.putBoolean(KEY_HAS_HTML, builder.isHtml());
+        if (builder.getMessage() != null) {
```

Comment:
I'm wondering if all of these null checks and stuff are necessary? I feel like we should still be able to keep it like before. 

```java
args.putString(KEY_TITLE, builder.getTitle());
args.putString(KEY_MESSAGE, builder.getMessage());
args.putString(KEY_BUTTON_TEXT, builder.getButtonText());
```

And then since this method would be pretty tiny and only used in one place, we can just put this back inline in the ```newInstance(Builder)``` method.